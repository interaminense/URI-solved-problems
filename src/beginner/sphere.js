/**
 * @author Adriano Interaminense <adriano.interaminense@gmail.com>
 *
 * Make a program that calculates and shows the volume of a sphere being provided the value of its
 * radius (R) . The formula to calculate the volume is: (4/3) * pi * R3. Consider (assign) for pi
 * the value 3.14159.
 *
 * INPUT
 *
 * The input contains a value of floating point (double precision).
 * @param {number} radius
 *
 * OUTPUT
 *
 * The output must be a message "VOLUME" like the following example with a space before and after
 * the equal signal. The value must be presented with 3 digits after the decimal point.
 */

const tmpl = result => `VOLUME = ${result.toFixed(3)}`;

const volume = radius => tmpl((4 / 3) * 3.14159 * (radius ** 3));

export { volume, tmpl };
