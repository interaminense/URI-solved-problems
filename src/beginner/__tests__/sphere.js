import { volume, tmpl } from '../sphere';

describe('Sphere', () => {
  let radius = 0;
  const result = R => volume(R);

  it('The volumn passin 3 must be 113.097', () => {
    radius = 3;

    expect(result(radius)).toEqual(tmpl(113.097));
    expect(result(radius)).toMatchSnapshot();
  });

  it('The volumn of 15 must be 14137.155', () => {
    radius = 15;

    expect(result(radius)).toEqual(tmpl(14137.155));
    expect(result(radius)).toMatchSnapshot();
  });

  it('The volumn of 1523 must be 14797486501.627', () => {
    radius = 1523;

    expect(result(radius)).toEqual(tmpl(14797486501.627));
    expect(result(radius)).toMatchSnapshot();
  });
});
