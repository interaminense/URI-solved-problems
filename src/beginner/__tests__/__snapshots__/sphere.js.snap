// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sphere The volumn of 15 must be 14137.155 1`] = `"VOLUME = 14137.155"`;

exports[`Sphere The volumn of 1523 must be 14797486501.627 1`] = `"VOLUME = 14797486501.627"`;

exports[`Sphere The volumn passin 3 must be 113.097 1`] = `"VOLUME = 113.097"`;
